c2rust-analyze/src/main.rs

```diff
@@ -849,6 +946,7 @@ impl rustc_driver::Callbacks for AnalysisCallbacks {
 
 fn main() -> rustc_interface::interface::Result<()> {
     init_logger();
+    panic::set_hook(Box::new(panic_detail::panic_hook));
```

Comment:
What happens if there is a panic outside out of the `panic::catch_unwind`s?  Then we don't get the normal panic that prints the error message and panics, right, but we want that?

I think we should either have a solution similar to the span guard, where we set and then remove the panic hook only during/immediately surrounding a `panic::catch_unwind`.  Or have a top-level `panic::catch_unwind` that carries out the default panic behavior.  Otherwise, we'll start getting silent panics, and that seems very bad.

Reply:
Good point.  I've fixed it to call the default panic hook if a panic occurs outside our `panic_detail::catch_unwind` wrapper.